universal-application-tool-0.0.1/app/services/program/ProgramServiceImpl.java

```diff
@@ -396,7 +396,10 @@ public void publishProgram(long id) throws ProgramNotFoundException {
   public ProgramDefinition newDraftOf(long id) throws ProgramNotFoundException {
     ProgramDefinition program =
         this.getProgramDefinition(id).toBuilder().setLifecycleStage(LifecycleStage.DRAFT).build();
-    return programRepository.insertProgramSync(program.toProgram()).getProgramDefinition();
+    Program programBean = program.toProgram();
+    programRepository.insertProgramSync(programBean);
+    programRepository.updateQuestionVersions(programBean);
```

Comment:
Question (non-blocking): why is it necessary to update question versions after creating a new program version if the questions are copy-on-write?

Reply:
Ah makes total sense, thank you